python/turbine_models/custom_models/stateless_llama.py

```diff
@@ -171,32 +184,32 @@ def forward(token0: torch.Tensor, *state0_flat):
             state0 = pytree.tree_unflatten(state0_flat, state_schema)
             result = mod.forward(token0, past_key_values=state0)
             state1_flat, _ = pytree.tree_flatten(result.past_key_values)
-            state1_flat = [torch.transpose(x[:, :, -1:, :], 1, 2) for x in state1_flat]
+            state1_flat = [
+                torch.transpose(x[:, :, -1:, :], 1, 2) for x in state1_flat
+            ]
             token1 = torch.argmax(result.logits[:, -1, :], dim=1)
             token1 = token1[None, :]
             return token1, *state1_flat
 
-    import_to = "IMPORT" if compile_to == "torch" else "INPUT"
+    import_to = "INPUT" if compile_to == "linalg" else "IMPORT"
     inst = StateUpdateModule(context=Context(), import_to=import_to)
 
     # TODO: Integrate with external parameters to actually be able to run
     # TODO: Make more generalizable to be able to quantize with all  compile_to options
-    if quantization == "int4" and compile_to == "torch":
+    if args.quantization == "int4" and not compile_to == "linalg":
```

Comment:
Why not run the group quant rewrite when compiling to linalg? I might be misunderstanding what this line is trying to predicate.

Reply:
please don't use args inside the function, makes it hard to use externally

Reply:
The MMGroupQuantRewriterPass currently works at the torch ir level (specifically matches against  torch.aten.mm). So I can run it compiling to linalg but it wouldn't actually accomplish anything

Reply:
> please don't use args inside the function, makes it hard to use externally

Good catch. Removed now

Reply:
I think I'm missing a detail here then. If we're compiling to linalg, do we not go through torch still? In which case the resulting linalg IR would have the group quant rewrite?

Reply:
ok, I understand now. Yeah that layering does sound a bit strange, but this check makes sense to me now.